runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -1418,7 +1418,13 @@ static EnvironmentInfo create(String environmentId, String containerUrl) {
 
   static void configureSdkHarnessContainerImages(
       DataflowPipelineOptions options, RunnerApi.Pipeline pipelineProto, Job newJob) {
-    if (useUnifiedWorker(options)) {
+
+    boolean isDataflowPrimeJob =
+        (options.getDataflowServiceOptions() != null
+            && options.getDataflowServiceOptions().contains("enable_prime"));
```

Comment:
The intention behind `--dataflow_service_options` is to have a flag that the SDK sort of promises not to inspect, though...

Reply:
Yeah, seems like we can just remove this check after a Dataflow service change. Will re-send this for review after the service change is fully deployed.